### ISSUE TITLE
Accessibility of places

### ISSUE THREAD

**Issue by chaals** [schemaorg/schemaorg#254]
We should provide some simple properties to describe the accessibility characterisitics of places (do restaurants have a wheelchair-accessible entry? What about toilet? Is there enough guidance for a blind person to find their way around a train station with a cane? Is there an assistance service available?…)

**Comment by danbri** [comment]
Thanks! Can you dig out our previous discussions and experiments? /cc @rvguha

**Comment by danbri** [comment]
http://lists.w3.org/Archives/Public/public-vocabs/2014Mar/0075.html

Here's my and @chaals versions of the strawman we hacked in London last year: 

* https://github.com/danbri/schemaorg/blob/sdo-physa11y/data/sdo-physa11y-schema.rdfa https://github.com/danbri/schemaorg/blob/sdo-physa11y/data/sdo-physa11y-examples.txt
* https://github.com/chaals/schemaorg/blob/sdo-physaxs/data/sdo-physa11y-schema.rdfa https://github.com/chaals/schemaorg/blob/sdo-physaxs/data/sdo-physa11y-examples.txt

@rvguha has argued that we should try harder to get agreement on machine-friendly controlled values e.g. around common usecases like finding a restaurant that suits wheelchairs, to enable search engines to offer search filters or prioritization.

To summarize ... in 2014 we got as far as rough consensus that we want to do something in this area, but couldn't agree on exactly what.

**Comment by chaals** [comment]
See also thread beginning https://lists.w3.org/Archives/Public/public-vocabs/2015Mar/0087.html

**Comment by westurner** [comment]
* https://www.w3.org/wiki/WebSchemas/Accessibility
* accessibilityFeature -- https://schema.org/accessibilityFeature
* accessibilityHazard -- https://schema.org/accessibilityHazard
* accessibilityAPI -- https://schema.org/accessibilityAPI
* accessibilityControl -- https://schema.org/accessibilityControl
* accessibilityPhysicalFeature

- [ ] P: an accessibilityPhysicalFeature property with a range including {Place, Event, Course, CourseInstance} and a domain TBD given one of the following solutions:

- OR
  - Old/current-school schema.org:
    - [ ] a wiki page like https://www.w3.org/wiki/WebSchemas/Accessibility listing enumerated Text values in a table
      - [ ] accessibilityPhysicalFeature domain: Text
    - [ ] Enumeration and Enumeration::members like https://schema.org/OrderStatus
      - [ ] accessibilityPhysicalFeature domain: AccessibilityPhysicalFeature
      - [ ] an Enumeration > AccessibilityPhysicalFeature [Enumeration](https://meta.schema.org/Enumeration) class
        - name
        - URL
        - description
        - \[image]
        - \[thumbnail]
       - [ ] AccessibilityPhysicalFeature:: Enumeration members
  - [ ] EnumerationValueSet, EnumerationValue: https://github.com/schemaorg/schemaorg/issues/894
    - [ ] a wiki page like https://www.w3.org/wiki/WebSchemas/Accessibility listing the EnumerationValue(s) as an EnumerationValueSet \["external enumerations"]
      - [ ] accessibilityPhysicalFeature domain: _?
    - [ ] an EnumerationValueSet of EnumerationValue(s) for accessibilityPhysicalFeature
      - [ ] accessibilityPhysicalFeature domain: _?

**Comment by westurner** [comment]
From https://github.com/schemaorg/schemaorg/pull/972#issuecomment-173698449 
- ADA symbols: https://www.access-board.gov/guidelines-and-standards/buildings-and-sites/about-the-ada-standards/ada-standards/chapter-7-communication-elements-and-features

From https://github.com/schemaorg/schemaorg/pull/972#issuecomment-175907591 : 
- Accessible tourism
  https://en.wikipedia.org/wiki/Accessible_tourism
- ADA signs
  https://en.wikipedia.org/wiki/ADA_Signs
- International Symbol of Access (wheelchair sign)
  https://en.wikipedia.org/wiki/International_Symbol_of_Access
- Downloadable Disability Access Symbols (12)
  https://www.graphicartistsguild.org/tools_resources/downloadable-disability-access-symbols

**Comment by westurner** [comment]
@danbri @rjw How should these Enumerations (?) be listed? https://github.com/schemaorg/schemaorg/issues/254#issuecomment-223049660

**Comment by mediaprophet** [comment]
Example pre-existing datasets for localGovernment related information: http://data.gov.au/organization/city-of-ballarat 

noting:

http://data.gov.au/dataset/ballarat-reserves
http://data.gov.au/dataset/ballarat-dog-bag-dispensers
http://data.gov.au/dataset/ballaratbbq
http://data.gov.au/dataset/ballarat-bike-racks
http://data.gov.au/dataset/ballarat-litter-bins
http://data.gov.au/dataset/ballarat-drinking-fountains
http://data.gov.au/dataset/ballarat-parking-machines
http://data.gov.au/dataset/ballarat-artwork
http://data.gov.au/dataset/ballarat-tables-and-seating
http://data.gov.au/dataset/ballarat-car-parking-areas
http://data.gov.au/dataset/ballarat-playgrounds
http://data.gov.au/dataset/ballarat-public-toilets
http://data.gov.au/dataset/ballarat-senior-citizen-clubs
http://data.gov.au/dataset/ballarat-aquatic-facilities
http://data.gov.au/dataset/ballarat-cultural-facilities
http://data.gov.au/dataset/ballarat-indoor-recreation-facilities
http://data.gov.au/dataset/ballaratbusshelters
http://data.gov.au/dataset/ballarat-avenue-of-honour-plaques
http://data.gov.au/dataset/ballarat-off-leash-areas

noting that whilst the datasets need to be made available in some suitable form of structured-data interface (perhaps userAgent aware, et.al.) i had difficulty finding suitable Schema.org references, where the objective i hoped to figure out is how to produce an accessibility pathway.

ie: 'ok google, where's the closest toilet'....

**Comment by danbri** [comment]
Note that the latest schema.org release (this week's 3.1) includes a general construction http://schema.org/amenityFeature + http://schema.org/LocationFeatureSpecification to capture a long tail of location-related features that are not explicitly anticipated by schema.org. 

We should still continue towards adding dedicated support but this is a useful extension mechanism, or possible prototyping syntax, that can complement more explicit markup structures.

**Comment by westurner** [comment]
> Note that the latest schema.org release (this week's 3.1) includes a general construction http://schema.org/amenityFeature + http://schema.org/LocationFeatureSpecification to capture a long tail of location-related features that are not explicitly anticipated by schema.org.

Use case: Specify that an Event (in a Place w/ "WheelchairRamps" (?)) will include sign language interpretation.

- In this example, there would be properties for/of the Event and for/of the Place

Thing > Intangible > StructuredValue > PropertyValue > LocationFeatureSpecification

PropertyValue > ?:
- value r: {Boolean, Number, StructuredValue, Text}
- valueReference r: {Enumeration, PropertyValue, QualitativeValue, QuantitativeValue, StructuredValue}
- url r:URL
- sameAs r:URL



> We should still continue towards adding dedicated support but this is a useful extension mechanism, or possible prototyping syntax, that can complement more explicit markup structures.

- Should there be Enumerations for IRL accessibilityFeatures?
  - I think there should.
    - Would an EnumerationSet be helpful? https://github.com/schemaorg/schemaorg/issues/254#issuecomment-223049660 (thanks @mediaprophet)
- With PropertyValue, the range of valueReference includes Enumeration.
- Is EnumerationSet the current/correct/most consistent way to specify known "AccessibilityPhysicalFeature"s?

```
Event
  name: speech #1
  accessibilityPhysicalFeature:
     - PropertyValue > ?:
       name: Sign Language Interpretor
       value: True
       url: <EnumerationSet::SignLanguageInterpretor>
  location:
    Place
      name: auditorium xyz
      amenityFeature:
        - LocationFeatureSpecification:
             name: "Off Leash Areas"
             value: True
             url: http://data.gov.au/dataset/ballarat-off-leash-areas
             valueReference: ?
      accessibilityPhysicalFeature:
         - PropertyValue > ?:
             name: Wheelchair Ramps
             value: True
             url: <EnumerationSet::WheelchairRamps>
             description: "..."

```

**Comment by westurner** [comment]
People:

- Domain experts
  - Which organizations have an expert to help out with this?
- Schema.org experts
  - @danbri
  - @rjw
- Schema.org amateurs
  - @westurner
  -
- Data publishers
  -
- Data consumers
  - Data presenters: e.g. search engines, aggregator sites

User stories:

- Data publishers can specify available accessibilityPhysicalFeatures (at least) for Places and Events (**URLs**)
- Data presenters can display (themed) icons for available services (**URLs**)
- Data consumers can locate Place and Events with specific AccessibilityPhysicalFeatures (**URLs**)

Technical approach:

- Enumeration > AccessibilityPhysicalFeature 
- List of Enumeration URLs for known "AccessibilityPhysicalFeatures"
  - @accessibilityExperts: catalog known accessibility features
  - @schemaorgExperts: should these be Enumerations in schema.org core or listed in a wiki page?
- Property: "accessibilityPhysicalFeature"? d: {Place,Event}
  - @schemaorgExperts: Relation to amenityFeature r: LocationFeatureSpecification

**Comment by mediaprophet** [comment]
Also note:
https://drive.google.com/file/d/0B_-AWWDVv3V2YWgzdkd2Vlo4ZG8/view?usp=sharing

WIP...

**Comment by mediaprophet** [comment]
Also, whilst noting https://github.com/schemaorg/schemaorg/issues/509 what about EV Charge Points?

**Comment by westurner** [comment]
Is the new Google Maps wheelchair accessibility information schema.org
JSONLD?
https://www.blog.google/products/maps/building-map-everyone/

**Comment by danbri** [comment]
We have failed to wrap this up in 4 years. A lot of that came before we had the "trying ideas out" area, Pending.schema.org.

I propose we simply add **to pending** a property isWheelchairAccessible that takes either or any of Text, Boolean, URL.  The text and URL could explain nuances. We also can explore http://schema.org/amenityFeature + http://schema.org/LocationFeatureSpecification .

Any objections?

**Comment by danbri** [comment]
Looking at https://www.localguidesconnect.com/t5/General-Discussion/Answering-accessibility-questions-on-Google-Maps/m-p/26#M2270 (which afaik has no JSON-LD connection), there are more detailed notions of "wheelchair accessible". How about we handle those as sub-properties?

* Is there a wheelchair-accessible entrance? -> hasWheelchairAccessibleEntrance
* Is there wheelchair-accessible seating? -> hasWheelchairAccessibleSeating
* Is there wheelchair-accessible parking? -> hasWheelchairAccessibleParking
* Is there a wheelchair-accessible elevator? -> hasWheelchairAccessibleElevator 

but all rolling up to isWheelchairAccessible as a general property? (the has/is is awkward, admittedly)

**Comment by vholland** [comment]
+1 to adding isWheelchairAccessible to pending to get this unstuck. If we see providers with more granular data, we can eventually move to an enum.

**Comment by danbri** [comment]
Noting that this issue tracks the broader case ("Is there enough guidance for a blind person to find their way around a train station with a cane? Is there an assistance service available?…"), I'd like to do something now for the wheelchair aspect while keeping things open to acknowledge that there is a large area of possible work here. I'll file a sub-issue for 'simple' (nothing's ever simple but...) wheelchair boolean into Pending.

**Comment by rvguha** [comment]
Dan, I could not agree more strongly. The last time around, we lost our way
in discussions about different levels of handicap access (are the toilets
big enough for wheelchairs, ...) and ended up doing nothing. Lets get
started with something useful and we can add to it later.

guha

**Comment by holgerd** [comment]
Great to see schemaorg is discussing accessibility of places.

I'm the co-founder of https://wheelmap.org and https://accessibility.cloud. Our non-profit organisation has started describing place accessibility definitions from dozens of different sources worldwide, bringing information about >1.5mio places together under one API. As a "Disabled People Organization" we are domain experts on this topic.

We developed and use the format http://a11yjson.org right now but eventually want to contribute to and learn from a larger standard. This is how I found out about this github issue. :)

This is my first post within the schema.org community, any advice on how to get involved is appreciated.

(cc @opyh)

**Comment by rvguha** [comment]
Holger,

  Welcome to schema.org! We have been hoping to introduce schema to enable
structured description about accessibility of places for a long time.
Participation from wheelmap would be great.

 Do you have a proposal for schema additions/changes to Schema.org?

guha

**Comment by k00ni** [comment]
Its great to see interest in place accessibility!

I started using the a11y-json format a while ago and been helping in their [issue tracker](https://github.com/sozialhelden/ac-format/issues) (e.g. add new properties, refine existing ones). IMHO they chose a way to combine expressiveness with easy usage and i would like to see this in schema.org. Talking to firms in the tourism sector, they all move in this direction and want to use such data.

**How to process further? How can we help here?**

**Comment by opyh** [comment]
Hi, I'm working with Holger at Sozialhelden e.V. - I'm currently sorting this issue a bit so we can apply a divide and conquer strategy, so I'd split this into sub-issues as next step.

For many other things in the real world, sets of more complex attributes are needed for a real description that fits the stakeholder needs.

I'd see the following issues/next steps where we need a consensus to go on:

1) Generally I'd follow a proposal that works analog to describing the accessibility of `CreativeWork`: For most stakeholders, physical accessibility features, hazards, and controls should be lists of defined terms so you can simply show them like other features of a place (e.g. a sauna of a hotel). Many physical features can simply be described in terms of their (non-)existence, analog to 'flashingHazard' as a value for `accessibilityHazard` for `CreativeWork`s.
 What do you think about extending `Thing` with `accessMode`, `accessModeSufficient`, `accessibilityAPI`, `accessibilityControl`, `accessibilityFeature`, `accessibilityHazard`, `accessibilitySummary` and allow these properties to refer to other `Thing`s? This would allow to describe more complex accessibility information for physical things in an understandable way that is open to associations with other data, so you could e.g. refer to a door opener switch from an external IoT vocabulary as `accessibilityControl` of a door, or an induction loop as feature of a concert hall. There could be a shared term vocabulary for the use case of discovery and search result filtering that could include terms like `"fullWheelchairAccessibility"`, `"audioDescription"` or `"signLanguage"`, and still it would support more complex extensions like `AnimalPolicy`, `Elevator`, extensions of `Room`, … for more complex data use cases. After this extension, `amenityFeature`/`LocationFeatureSpecification` would inherit e.g. `accessibilityFeature`, which would allow to specify location features as other `Thing`s.
3) We need a set of types to describe physical / civic / architectural features of places that facilitate mobility in general, e.g. stairways, travelators, escalator, elevators, doors that connect `Place`s and a separate concept of entrances/exits (an entrance is not necessarily a door, but can be an escalator, for example). Data for this already exists on OpenStreetMap, Wheelmap, and in GTFS patways, for example.
4) We need to be able to describe sanitary facilities like `Toilet`s and `Bathroom`s.

So far, we've identified the following domains:

- Architecture, spatial modeling and Building Information Management (BIM): Architects and designers need a standardized means to exchange data about units like rooms. Their perspective on a `House` is probably its physically nested structure, its rooms, spaces and connections between these spaces. Accessibility relevant items would be entrances, exits, doors, elevators, toilets, or ground surfaces inside and around a building. Ultimately, they should be able to convert data from and to BIM data formats, e.g. to export BIM data to use it for providing an online service. An example of this use case is Apple IMDF: This format allows BIM modelers to export a data archive that Apple Maps can use for indoor routing. Being a central ‘source of truth’ for managing building data, BIM tools should be a place where accessibility data is managed, too. A wheelchair-accessible toilet, for example, would not only be modeled spatially, but also as a `Product` bundle with a `3DModel`, a GTIN code, a manufacturer, an owner, an operator, a price, and `Offer`s for replacing it when it’s broken. Hotels can be modeled with BIM too, allowing for an automated export of their `Room`s, `AmenityFeature`s, and their accessibilities into offer listing software.
- Transport and Navigation: A physical place is not accessible if it’s not possible to navigate there without barriers. For accessible mobility we need some level of compatibility to external vocabularies and data formats that focus on navigation, in most cases by finding an optimal route through a connected graph. This proposal contains entities that can describe an entry/exit of a `Place`. As we are accessibility experts, but not routing algorithm experts, we want to leave a solution for data model interoperability to domain specialists like the authors of LinkedGTFS, and add accessibility-related properties to their proposed types at a later stage.
- Booking a service online (e.g. a hotel): To enable discovery of accessible services online, booking platforms must have a simple means to add searchable criteria to places and services at these places. In the STI ontology, it’s already possible to add amenity features like a sauna, but standardized terms for accessibility features like a hoist in a swimming pool are yet to be defined. An important issue here are quantity constraints: A frequent use case is to search for a minimum constraint for an installed equipment - like searching for restaurants that are only accessible for an individual wheelchair user if all doors are >90cm wide. Quantity constraints can already be expressed in schema.org, but we need to define a best practice how to declare and crawl this information for amenity features to enable real world use of it.
- Physical product accessibility from a user’s perspective: Physical `Thing`s, for example `ProductModel`s, can have aspects that make them (in-)accessible to their users. To describe this, we could add properties to `Action` (e.g. to describe that to use a specific phone booth, you have to be able to use both hands at a height of 100cm from the floor), or to the `Thing`/`Product` /`ProductModel` itself, analog to the accessibility description means of `CreativeWork`. Additionally, it should be possible to define a personal profile to match products and media to a user’s needs. A proposal for this exists already in schema.org GitHub issue #1997. Additionally, services bound to a physical location can have facilities built to make the service available for people with disabilities. Facility types and individual facilities must be representable in the vocabulary - as amenity features, and as physical tangible things at a spatial location. We propose to connect the concept of `amenityFeature`, `Thing` and `Place` for this. Examples for this are braille sitemaps (which can are already be described with existing means in schema.org), toilet features like handlebars, the presence of staff that is trained in a sign language variant, or induction loops in concert halls.

Here is a list of example accessibility cases that should be describable:

- A hotel booking offer for a hotel that wants to advertise a specific accessibility feature discoverable by search
- A restaurant that is completely wheelchair accessible, but has no wheelchair accessible toilet
- Two rooms with a shared door used as entrance/exit for both rooms
- A supermarket that is connected to the next sidewalk with one main entrance door and two separate exits
- A restaurant entrance that is an escalator
- An elevator that has to be used to enter/exit a house
- A hotel with a suite with two bathrooms, of which one has a wheelchair accessible toilet, wash basin and mirror
- A subway station that has handrails and turnstiles for checking tickets that can only be used one-way
- A train platform inside a station that has tactile guide strips
- A concert hall with an induction loop
- A swimming pool with a hoist
- A beach with rentable beach wheelchairs
- A guided museum tour with two offers, of which one is in sign language, at specified times, and for free
- A public toilet that can only be opened with a Euro key

**Comment by tguild** [comment]
We've launched a related community group at W3C
https://www.w3.org/community/lda/

In W3C Automotive and Transportation Business Group, started a task force coordinating between ISO ITS, ISO SmartCities, OGC, W3C and others around transportation ontologies. We plan on following Spatial Data on the Web Best Practices and bring core ontologies to schema.org, the modular compound ones like VSSo will use those 

@chaals and @danbri would appreciate syncing up with you too, sadly not into the wee hours in some random part of the world

**Comment by westurner** [comment]
Bump

**Comment by westurner** [comment]
Are #39 and #41 the current issues for this feature?

Given a PR for #39 and/or #41, would it be possible for sites to include ADA accessibility symbols (see: https://github.com/schemaorg/schemaorg/issues/254#issuecomment-230009264 ) on Place, LocalBusiness, and Events?

**Comment by epaulson** [comment]
Here are some of the Google attributes for accessibility: https://support.google.com/business/answer/7298639?hl=en 

Unfortunately it looks like you set them in the Google Business listings, and not in any schema.org markup
= = = = 
Wheelchair-accessible entrance
Answer yes if the entrance to the business is approximately 3 feet wide and doesn’t have steps. 3 feet (or 1 meter) is approximately the minimum comfortable width for two people to stand right next to each other. If there are one or more steps, there should be a permanent ramp or, at the very least, a moveable ramp. Entryways that consist of only revolving doors should be marked No for this attribute.

Wheelchair-accessible restroom
Answer yes if the entrance to the restroom is at least one meter wide and can be reached without going up or down steps. If a person in a wheelchair would need to enter a stall inside the restroom, the stall’s entrance also needs to be one meter wide. (Remember, one meter is about the width of two people standing comfortably side by side.)

Wheelchair-accessible seating
Answer yes if the main area of the business can be accessed without stairs and there’s enough space for someone in a wheelchair to navigate to and sit at a table. If all tables are high (e.g. at standing level), the business isn’t wheelchair-friendly.

Wheelchair-accessible parking
Answer yes if there’s a parking spot specifically marked for those with accessibility needs. These spots are often marked with specific painting on the ground, placards, or signs (depending on your country and/or region).

Wheelchair-accessible elevator
Answer yes if a place has multiple floors and there is an elevator that is large enough to accommodate a wheelchair.
